### PR TITLE
RELATED: ONE-3252 ONE-3244 ONE-3246 Heatmap legend fixes

### PR DESCRIPTION
**Pull request by peterlacko**

### CODE REVIEW

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/676/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/253/ : SUCCESS in 5m 20s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/127/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/677/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/26/ : UNSTABLE in 37m 09s

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 1]
```diff
@@ -1259,7 +1259,12 @@ export function getHeatMapDataClasses(series: any = [], colorPalette: string[]):
         borderWidth: 0
     }));
 
-    const values: number[] = compact(get(newSeries, '0.data', []).map((item: any) => item.value));
+    const values: number[] = without(get(newSeries, '0.data', []).map((item: any) => item.value), null, undefined, NaN);
```
Comment (pbenes) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: I'm not sure if this would work, see `_.without(array, [values])`

Comment (pbenes) [reply src/components/visualizations/chart/chartOptionsBuilder.ts]: Ah, my bad, I see it exactly that way in examples

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/688/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/258/ : SUCCESS in 1m 31s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/141/artifact/test-storybook/html-report/index.html

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/144/artifact/test-storybook/html-report/index.html

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/146/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/689/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/45/ : UNSTABLE in 10m 06s

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/152/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/700/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/50/ : SUCCESS in 7m 32s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/138/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/273/ : SUCCESS in 2m 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/159/ : SUCCESS in 8m 03s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/29/ : SUCCESS in 13m 24s

**Pull request merged**